### PR TITLE
update(code): adding rbac add new role and fetch roles funtionalities

### PR DESCRIPTION
**Pull request by rohanranjan250613**

### CODE REVIEW

File: packages/sdk/src/modules/rbac/index.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+export class RBAC extends Base {
+  constructor(
+    protected override readonly _api?: ApiPromise,
+    protected readonly _metadata?: SDKMetadata
+  ) {
+    super();
+  }
+  /**
+   * Creates a new role.
+   * @param name - The options for creating the Roles.
```
Comment (irediaes) [inline packages/sdk/src/modules/rbac/index.ts]: wrong param name: "name" should be changed to "options"

File: packages/sdk/src/modules/rbac/index.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        id: id || generatedId,
+      };
+    } catch (error) {
+      throw new Error(`Error creating role: ${error}`);
+    }
+  }
+
+  /**
+   * Fetch all roles.
+   * @param name - The ownerAddress is public address of user or owner who created a roles.
```
Comment (irediaes) [inline packages/sdk/src/modules/rbac/index.ts]: wrong param name: "name" should be changed to "ownerAddress"

Please check all your codes for this and correct them.

File: packages/sdk/src/modules/rbac/index.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        let payload = {
+          id,
+          name,
+          enabled,
+        };
+        responseData.push(payload);
+      });
+      return responseData;
+    } catch (error) {
+      throw new Error(`Error while fetchroles: ${error}`);
```
Comment (irediaes) [inline packages/sdk/src/modules/rbac/index.ts]: > Error while fetchroles:

"Error occurred while fetching roles"

use the above error message instead

**Review by irediaes — CHANGES_REQUESTED** [review]
Code looks good.
Few typos check attached comments.

File: packages/sdk/src/modules/rbac/index.ts (hunk 1 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/rbac/index.ts]: done

File: packages/sdk/src/modules/rbac/index.ts (hunk 2 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/rbac/index.ts]: done

File: packages/sdk/src/modules/rbac/index.ts (hunk 3 above)
Comment (rohanranjan250613) [reply packages/sdk/src/modules/rbac/index.ts]: done

**Review by irediaes — APPROVED** [review]

**Pull request merged**